src/components/Offers/Form/form-components/OfferForm.js

```diff
@@ -0,0 +1,378 @@
+import {
+    CardContent,
+    CardHeader,
+    DialogContent,
+    FormHelperText,
+    Grid,
+    FormControl,
+    Typography,
+    Collapse,
+    Button,
+} from "@material-ui/core";
+import React, { useState, useCallback, useContext } from "react";
+import { Redirect } from "react-router-dom";
+import PropTypes from "prop-types";
+import MultiOptionTextField from "../../../utils/form/MultiOptionTextField";
+import MultiOptionAutocomplete from "../../../utils/form/MultiOptionAutocomplete";
+import ConnectedLoginAlert from "../../../utils/LoginAlert";
+import TitleComponent from "./TitleComponent";
+import OwnerComponent from "./OwnerComponent";
+import LocationComponent from "./LocationComponent";
+import JobTypeComponent from "./JobTypeComponent";
+import JobStartDateComponent from "./JobStartDateComponent";
+import JobDurationComponent from "./JobDurationComponent";
+import VacanciesComponent from "./VacanciesComponent";
+import IsPaidComponent from "./IsPaidComponent";
+import PublicationDateComponent from "./PublicationDateComponent";
+import PublicationEndDateComponent from "./PublicationEndDateComponent";
+import IsHiddenComponent from "./IsHiddenComponent";
+import TextEditorComponent from "./TextEditorComponent";
+import {
+    KeyboardArrowDown,
+    KeyboardArrowUp,
+} from "@material-ui/icons";
+import useOfferFormStyles from "./offerStyles";
+import { Controller } from "react-hook-form";
+import { useMobile } from "../../../../utils/media-queries";
+import "../editor.css";
+
+export const PAID_OPTIONS = [
+    { value: "none", label: "Unspecified" },
+    { value: true, label: "Paid" },
+    { value: false, label: "Unpaid" },
+];
+
+const OfferForm = ({ context, title }) => {
+    const {
+        submit,
+        errors,
+        requestErrors,
+        control,
+        fields,
+        contacts,
+        requirements,
+        getValues,
+        appendContact,
+        removeContact,
+        appendRequirement,
+        removeRequirement,
+        fieldsSelectorProps,
+        techSelectorProps,
+        loading,
+        success,
+        offerId,
+        isAdmin,
+        isLoggedIn,
+        companyUnfinishedRegistration,
+        showCompanyField,
+        showHiddenField,
+    } = useContext(context);
+
+    const isMobile = useMobile();
+
+    const formDisabled = !isLoggedIn || companyUnfinishedRegistration;
+
+    const [isAdvancedOpen, setAdvancedOpen] = useState(false);
+
+    const isAdvancedOpenOrErrors = useCallback(() => (
```

Comment:
The code already tells you that, better choose a name that gives more information

```suggestion
    const shouldShowAdvancedOptions = useCallback(() => (
```